addon/ng2/commands/newroute.ts

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+import * as path from 'path';
+import * as fs from 'fs';
+import * as Promise from 'ember-cli/lib/ext/promise';
+import * as Blueprint from 'ember-cli/lib/models/blueprint';
+import * as EmberGenerateCommand from 'ember-cli/lib/commands/generate';
+import * as SilentError from 'silent-error';
+import * as _ from 'lodash';
+
+var EOL = require('os').EOL;
+const chalk = require('chalk');
+
+const newRouteCommand = EmberGenerateCommand.extend({
```

Comment:
What does extending from `EmberGenerateCommand` do, vs just extending `Command` directly? i see you're overriding the `run` method so I suspect you may not be using the `EmberGenerateCommand` functionality at all.